blocks/components/editable/index.js

```diff
@@ -263,11 +233,11 @@ export default class Editable extends wp.element.Component {
 		}
 
 		content = wp.element.renderToString( content );
-		this.editor.setContent( content );
+		this.editor.setContent( content, { format: 'raw' } );
```

Comment:
This breaks merging a paragraph into a heading, because this ignore the content sanitization (due to the custom parent node). 

The behaviour of this branch appends a `p` element at the end of the Heading and thus transforms the heading into a multi-paragraph block. We could solve this by dropping the `p` container in the merge function, but I feel this validation is important to ensure each block contain only what we want it to contain.